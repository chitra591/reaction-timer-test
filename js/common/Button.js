import React, {Component} from 'react';
import {TouchableOpacity, Text, StyleSheet} from 'react-native';
import {stringConstants} from './constants';

export default class Button extends Component {
    constructor(props) {
        super(props);
    }
    onPressHandler = () => {
        const {onPress} = this.props;
        onPress();
    }
    render() {
        const {label, onPress, primary} = this.props;

        return(
            <TouchableOpacity 
                style={[styles.button, {backgroundColor: primary ?  stringConstants.YELLOW :  stringConstants.LIGHT_YELLOW}]}
                onPress={this.onPressHandler}>
                <Text style={styles.text}>
                    {label}
                </Text>
            </TouchableOpacity>
        );
    }
};

const styles = StyleSheet.create({
    button: {
        justifyContent: 'center', 
        height: '10%', 
        width: '80%', 
        justifyContent: 'center', 
        alignItems: 'center', 
        margin: 20, 
        borderWidth: 1, 
        borderRadius: 8, 
        borderColor: stringConstants.ASH
    },
    text: {color: stringConstants.GREY, textAlign: 'center'}
})