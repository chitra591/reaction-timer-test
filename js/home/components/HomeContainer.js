import {connect} from 'react-redux';
import HomeScreen from './HomeScreen';
import {getRandomNumber, setScore, clearScoreCard} from '../actions';

const mapStateToProps = (state, ownProps = {}) => {
    return state;
};

const mapDispatchToProps = (dispatch) => {
    return {
        getRandomNumber: (callback) => {
            dispatch(getRandomNumber(callback))
        },
        setScore: ({score}) => {
            dispatch(setScore({score}))
        },
        clearScoreCard: () => {
            dispatch(clearScoreCard())
        }
    };
};

const HomeContainer = connect(mapStateToProps, mapDispatchToProps)(HomeScreen);

export default HomeContainer;