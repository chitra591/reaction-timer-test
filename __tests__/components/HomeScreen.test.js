import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import React from 'react';
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import HomeContainer from '../../js/home/components/HomeContainer';
import Button from '../../js/common/Button';

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initialState = {
    scoreCard: {
        scores: [],
        average: 0.00
    }
}

describe('Testing HomeScreen', () => {
    it('renders as expected', () => {
        const wrapper = shallow(
            <HomeContainer store={mockStore(initialState)} />,);
        expect(wrapper.dive()).toMatchSnapshot();
    });
});

describe('Text', () => {
    describe('Rendering', () => {
        it('should match to snapshot - Primary', () => {
            const component = shallow(<Button label={"test label"} primary />)
            expect(component).toMatchSnapshot("Primary button snapshot")
        });
        it('should match to snapshot - Secondary', () => {
            const component = shallow(<Button label={"test label"} primary={false} />)
            expect(component).toMatchSnapshot("Secondary button snapshot")
        });
    });
    describe('Interaction', () => {
        describe('onPressHandler', () => {
            it('should call onPress', () => {
                const mockOnPress = jest.fn();      
                const component = shallow(<Button 
                    label= "test label" 
                    onPress={mockOnPress}       
                />);
                const instance = component.instance();  
                
                instance.onPressHandler();         
                expect(mockOnPress).toHaveBeenCalled();
                expect(mockOnPress).toHaveBeenCalledTimes(1);   
            });
        });
    });
});