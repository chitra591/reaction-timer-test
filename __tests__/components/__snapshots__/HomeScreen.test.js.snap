// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomeScreen renders as expected 1`] = `
<HomeScreen
  clearScoreCard={[Function]}
  getRandomNumber={[Function]}
  scoreCard={
    Object {
      "average": 0,
      "scores": Array [],
    }
  }
  setScore={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`Text Rendering should match to snapshot - Primary: Primary button snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#BFBFBF",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": "10%",
        "justifyContent": "center",
        "margin": 20,
        "width": "80%",
      },
      Object {
        "backgroundColor": "#a0fd98",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#767676",
        "textAlign": "center",
      }
    }
  >
    test label
  </Text>
</TouchableOpacity>
`;

exports[`Text Rendering should match to snapshot - Secondary: Secondary button snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#BFBFBF",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": "10%",
        "justifyContent": "center",
        "margin": 20,
        "width": "80%",
      },
      Object {
        "backgroundColor": "#c7ffc2",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#767676",
        "textAlign": "center",
      }
    }
  >
    test label
  </Text>
</TouchableOpacity>
`;
